lib/internal/fs/linux_recursive_watcher.js

```diff
@@ -0,0 +1,154 @@
+'use strict';
+
+const { EventEmitter } = require('events');
+const path = require('path');
+const { Symbol, ObjectKeys } = primordials;
+
+const kFSWatchStart = Symbol('kFSWatchStart');
+
+let internalSync;
+let internalPromises;
+
+function lazyLoadFsPromises() {
+  internalPromises ??= require('fs/promises');
+  return internalPromises;
+}
+
+function lazyLoadFsSync() {
+  internalSync ??= require('fs');
+  return internalSync;
+}
+
+async function traverse(dir, files = {}) {
+  const { stat, readdir } = lazyLoadFsPromises();
+
+  files[dir] = await stat(dir);
+
+  try {
+    const directoryFiles = await readdir(dir);
```

Comment:
Yea, we shouldn't use `readdir()` in any new core APIs. Please use `opendir()` and company.

Reply:
Thanks for the review. I just updated the pull request.